insights/combiners/satellite_version.py

```diff
@@ -95,30 +107,36 @@ def satellite_version(local, shared):
     information.
 
     """
-    # For Satellite 6.1.x (satellite_version/version.rb is available)
+    # For Satellite 6.1.x, if satellite_version/version.rb is available:
     sat6_ver = shared.get(Sat6Ver)
     if sat6_ver and sat6_ver.version:
         return SatelliteVersion(sat6_ver.full, sat6_ver.version, None,
                                 sat6_ver.major, sat6_ver.minor)
 
     rpms = shared.get(InstalledRpms)
     if rpms:
-        # For Satellite 6.2.x
+        # For Satellite 6.2.x, check the ``satellite`` package directly
         sat62_pkg = rpms.get_max('satellite')
         if sat62_pkg:
             version = sat62_pkg.version
             major, minor = _parse_sat_versoin(version)
             return SatelliteVersion(sat62_pkg.package, version,
                                     sat62_pkg.release, major, minor)
 
-        # For Satellite 6.0.x/6.1.x
-        fman_pkg = rpms.get_max('foreman')
-        sat6_ver = foreman_sat6_ver_map.get(fman_pkg.version) if fman_pkg else None
-        if sat6_ver:
-            major, minor = _parse_sat_versoin(sat6_ver)
-            return SatelliteVersion(sat6_ver, sat6_ver, None, major, minor)
-
-        # For Satellite 5.0
+        # For Satellite 6.0.x/6.1.x, check the version of:
+        # - foreman, candlepin and katello
+        fman = rpms.get_max('foreman')
+        canp = rpms.get_max('candlepin')
+        ktlo = rpms.get_max('katello')
+        if fman and canp and ktlo:
+            for sat_ver, map_ver in sat6_ver_map.items():
+                if (fman.version.startswith(map_ver.foreman) and
+                        canp.version.startswith(map_ver.candlepin) and
+                        ktlo.version.startswith(map_ver.katello)):
```

Comment:
maybe we can simplify 133 to 135 by below code:
  if all([a.startswith(b) for a,b in zip([fman, canp, ktlo], map_ver.values())]):


In further, if just use tuple as value in sat6_ver_map, We can replace map_ver.values() with map_ver.

Reply:
Cool point! will do it